tensorflow/core/common_runtime/rendezvous_mgr.cc

```diff
@@ -65,10 +65,10 @@ void IntraProcessRendezvous::SameWorkerRecvDone(
     StatusCallback done) {
   // Do a quick copy (sharing the underlying buffer) if both tensors
   // are on host memory.
-  const bool src_host =
-      (send_args.alloc_attrs.on_host() || parsed.src.type == "CPU");
-  const bool dst_host =
-      (recv_args.alloc_attrs.on_host() || parsed.dst.type == "CPU");
+  const bool src_host = (send_args.alloc_attrs.on_host() ||
+                         parsed.src.type == "CPU" || parsed.src.type == "SYCL");
```

Comment:
This also seems like a bug, btw.  This is saying that device memory and host memory are the same for SYCL, when it comes to memcopying.  E.g., that when you are copying from say, /CPU:0 to /device:SYCL:0, the tensor being pointed to is in CPU memory.  Is that intended?